From e67385ccb14dce7c5e19495e0c9898d482d9a0bb Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 26 Jul 2021 19:07:47 +0300
Subject: [PATCH] [SW-50527] gaudi: security: enable dbg protection

The DBG space follows the CFG space and has the prefix of 0x7FFF000000,
while the CFG is 0x7FFC000000.
To be able to address both memory spaces with the same security
framework functions, extend the PAGE_BASE mask to 2 additional bits.
0xFFF000 => 0x3FFF000

Change-Id: I22b68c3600d9963469914d9722470d5aa513b8d7
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi/arm/security.c b/arch/gaudi/arm/security.c
index 7c904f5..e1042c5 100644
--- a/arch/gaudi/arm/security.c
+++ b/arch/gaudi/arm/security.c
@@ -89,7 +89,7 @@
  */
 
 /* Protection bits helpers */
-#define PAGE_BASE(x)		((x) & 0xFFF000)
+#define PAGE_BASE(x)		((x) & 0x3FFF000)
 #define PROT_REG_BASE(x, off)	((PAGE_BASE(x)) + (off))
 #define PROT_REG_OFF(x)		(((x) & 0xF80) / 32)
 #define PRIV_PROT_REG(x)	(PROT_REG_BASE((x), 0xE80) + PROT_REG_OFF(x))
